### PR TITLE
Add storages and files to migration

### LINKED ISSUE #1985: Add file and storage resource types to the migration tool logic

### PR DESCRIPTION
**Pull request by imsdu**
Fixes #1985

### CODE REVIEW

File: delta/migration/src/main/scala/ch/epfl/bluebrain/nexus/migration/Migration.scala [hunk 1]
```diff
@@ -416,24 +427,26 @@ final class Migration(
...
-            case Created(_, _, _, _, types, _, _, _) if types.exists(viewTypes.contains)    =>
+            case Created(_, _, _, _, types, _, _, _) if types.exists(viewTypes.contains)          =>
               IO.pure(RunResult.Success)
-            case Updated(_, _, _, _, types, _, _, _) if types.exists(viewTypes.contains)    =>
+            case Updated(_, _, _, _, types, _, _, _) if types.exists(viewTypes.contains)          =>
               IO.pure(RunResult.Success)
-            case Deprecated(_, _, _, _, types, _, _) if types.exists(viewTypes.contains)    =>
+            case Deprecated(_, _, _, _, types, _, _) if types.exists(viewTypes.contains)          =>
               IO.pure(RunResult.Success)
             //TODO Storages
```
Comment (umbreak) [inline delta/migration/src/main/scala/ch/epfl/bluebrain/nexus/migration/Migration.scala:439]: todo not necessary anymore

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala [hunk 2]
```diff
@@ -45,8 +47,12 @@ import com.typesafe.scalalogging.Logger
 import fs2.Stream
 import monix.bio.{IO, Task, UIO}
 import monix.execution.Scheduler
-import retry.CatsEffect._
-import retry.syntax.all._
+import _root_.retry.CatsEffect._
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala:50]: This _root_ is not needed

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala [hunk 3]
```diff
@@ -544,6 +551,87 @@ final class Files(
     }
   }
 
+  override def migrate(
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala:554]: I find it quite disturbing the migration code is creeping all over the rest of the code. I would much rather have it composable not inherited like this.

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala [hunk 4]
```diff
@@ -544,6 +551,87 @@ final class Files(
...
+                                 MigrateFile(id, projectRef, storageRef, storage.tpe, attributes15, rev.getOrElse(0L), caller),
+                                 project
+                               ).void.mapError(MigrationRejection(_))
+    } yield res
+
+  override def fileAttributesUpdated(id: Iri, projectRef: ProjectRef, rev: Long, attributes: StorageFileAttributes)(
+      implicit subject: Subject
+  ): IO[MigrationRejection, Unit] = {
+    val digest =
+      if (attributes.digest == kg.Digest.empty) NotComputedDigest
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala:598]: You seem to have this logic in more than one place. Why not refactoring in a `private def` ?

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala [hunk 5]
```diff
@@ -769,12 +857,25 @@ object Files {
       case s: Current                   => IOUtils.instant.map(FileDeprecated(c.id, c.project, s.rev + 1L, _, c.subject))
     }
 
+    def migrate(c: MigrateFile) = state match {
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala:860]: Same here, the migration code is being entangled with the production regular code

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/FileCommand.scala [hunk 6]
```diff
@@ -114,4 +114,14 @@ object FileCommand {
     */
   final case class DeprecateFile(id: Iri, project: ProjectRef, rev: Long, subject: Subject) extends FileCommand
 
+  //TODO: Remove after migration
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/model/FileCommand.scala:117]: Same here

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala [hunk 7]
```diff
@@ -429,6 +431,26 @@ final class Storages private (
 
   private def identifier(project: ProjectRef, id: Iri): String =
     s"${project}_$id"
+
+  override def migrate(id: Iri, projectRef: ProjectRef, rev: Option[Long], source: Json)(implicit
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/Storages.scala:435]: same here

File: delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageCommand.scala [hunk 8]
```diff
@@ -89,4 +89,14 @@ object StorageCommand {
     */
   final case class DeprecateStorage(id: Iri, project: ProjectRef, rev: Long, subject: Subject) extends StorageCommand
 
+  //TODO: Remove after migration
```
Comment (umbreak) [inline delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/model/StorageCommand.scala:92]: same here

**Review by umbreak — CHANGES_REQUESTED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Review by umbreak — APPROVED** [review]

**Pull request merged**